nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/CommonClassLoaderServiceImpl.java

```diff
@@ -146,6 +146,11 @@ private void createCommonClassLoader() {
             Collections.addAll(cpElements,
                     domainLib.listFiles(new JarFileFilter()));
         }
+        final File domainLibExt = new File(domainDir, "lib/ext/"); // NOI18N
+        if (domainLibExt.isDirectory()) {
+            Collections.addAll(cpElements,
+                    domainLibExt.listFiles(new JarFileFilter()));
+        }
```

Comment:
Since the specific functioncality we depend on depends on #5649 (amazing, a permutation, of this pr ID #5694 ! :)) This is likely not the way to go, it would create two classloader hierarchies with incompatible classes. The requirement to extend system classloader still holds, otherwise eclipselink oracle extension will not work.

If I understood right, your problem was that these classes weren't found in application under such circumstances, which is possible because 

I think the solution is to change `com.sun.enterprise.v3.server.APIClassLoaderServiceImpl#getExtensionClassLoader` instead. I believe under these circumstances we should go for system classloader as parent of API classloader, or the best approach might be to target system classloader of Felix (bundle id 0), which will even respect boot delegation properties.